test/StringDiagram.lean

```diff
@@ -0,0 +1,56 @@
+import Mathlib.Tactic.Widget.StringDiagram
+import ProofWidgets.Component.Panel.SelectionPanel
+import ProofWidgets.Component.Panel.GoalTypePanel
+
+/-! ## Example use of string diagram widgets -/
+
+section MonoidalCategory
+
+open ProofWidgets
+
+open CategoryTheory
+open scoped MonoidalCategory
+
+universe v u
+
+variable {C : Type u} [Category.{v} C] [MonoidalCategory C]
+
+lemma left_triangle {X Y : C} (η : 𝟙_ _ ⟶ X ⊗ Y) (ε : Y ⊗ X ⟶ 𝟙_ _) (w : False) :
+    η ▷ X ≫ (α_ _ _ _).hom ≫ X ◁ ε = (λ_ _).hom ≫ (ρ_ _).inv := by
+  with_panel_widgets [SelectionPanel]
+    exact w.elim
+
+lemma yang_baxter {V₁ V₂ V₃ : C} (R : ∀ V₁ V₂ : C, V₁ ⊗ V₂ ⟶ V₂ ⊗ V₁) (w : False) :
+    R V₁ V₂ ▷ V₃ ≫ (α_ _ ..).hom ≫ _ ◁ R _ _ ≫ (α_ _ ..).inv ≫ R _ _ ▷ _ ≫ (α_ _ ..).hom =
+    (α_ _ ..).hom ≫ V₁ ◁ R V₂ V₃ ≫ (α_ _ ..).inv ≫ R _ _ ▷ _ ≫ (α_ _ ..).hom ≫ _ ◁ R _ _ := by
+  with_panel_widgets [GoalTypePanel]
```

Comment:
Instead of writing `with_panel_widgets` everywhere, you can also use a single `show_panel_widgets [local GoalTypePanel]` command towards the top of the file.